### ISSUE TITLE
Using Media Image custom attribute type  could not display on frontend. 

### ISSUE THREAD

**Issue by nimitaemipro** [magento/magento2#19054]
Using Media Image custom attribute type could not display on frontend.
### Preconditions (*)

1. Magento Community Edition V2.2.6 with sample data (PHP 7.0) 

### Steps to reproduce (*)

1. Create custom attribute named 'icon_image' of product from admin.
2. Set attribute type Media Image.
3. assign icon_image attribute in attribute set.
4. Update product image and set new role 'icon_image'.
5. Create custom code for displaying image in front end. code is like: 

**Sample code: 1**
```
$productImageAttr = $product->getCustomAttribute( 'icon_image' );
$productImage = $this->helper('Magento\Catalog\Helper\Image')
->init($product, 'icon_image')
->setImageFile($productImageAttr->getValue());

```
` <img src="<?php echo $productImage->getUrl() ?>" alt="<?php echo $block->escapeHtml($product->getTitle()) ?>" />`

**Sample code:2**
```
$objectManager = \Magento\Framework\App\ObjectManager::getInstance();
$mediaUrl =  $objectManager->get('Magento\Store\Model\StoreManagerInterface')
             ->getStore()
             ->getBaseUrl(\Magento\Framework\UrlInterface::URL_TYPE_MEDIA); 
$diamondIcon=$product->getResource()->getAttribute('diamond_icon')->getFrontend()->getValue($product);
```

` <img src=" <?php echo $mediaUrl."catalog/product".$diamondIcon ?>" />`

### Expected result (*)

1. Custom image / media type custom attribute  should be display on frontend.

### Actual result (*)

1. Custom image / media type custom attribute is not displayed  on frontend anymore.

**Comment by magento-engcom-team** [comment]
Hi @nimitaemipro. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@nimitaemipro do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @Nazar65. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by Nazar65** [comment]
I'm will work on this issue

**Comment by VladimirZaets** [comment]
Hi @nimitaemipro. Thank you for your report. 
The issue has been fixed in magento/magento2#19068 by @Nazar65 in 2.2-develop branch
Related commit(s):
 - [e3134ea23a4827e8d3ef0ed2b4027b66e565f215](https://github.com/magento/magento2/commit/e3134ea23a4827e8d3ef0ed2b4027b66e565f215)

The fix will be available with the upcoming 2.2.8 release.

**Comment by Nazar65** [comment]
Hi @nimitaemipro this issue occurs only in 2.2-develop branch, in 2.3-develop 
 this works fine.

**Comment by magento-engcom-team** [comment]
Hi @nimitaemipro. Thank you for your report. 
The issue has been fixed in magento/magento2#20096 by @davidverholen in 2.3-develop branch
Related commit(s):
 - [0edcd206c7ed18a3cae4332488d1cf52a877d181](https://github.com/magento/magento2/commit/0edcd206c7ed18a3cae4332488d1cf52a877d181)
 - [9aa1aebc0647c85670b58bca7ca8191938016c9c](https://github.com/magento/magento2/commit/9aa1aebc0647c85670b58bca7ca8191938016c9c)

The fix will be available with the upcoming 2.3.1 release.
